daemon/list.go

```diff
@@ -133,7 +133,18 @@ func (daemon *Daemon) Containers(config *ContainersConfig) ([]*types.Container,
 			ID:    container.ID,
 			Names: names[container.ID],
 		}
-		newC.Image = container.Config.Image
+
+		img, err := daemon.Repositories().LookupImage(container.Config.Image)
+		if err != nil {
+			// If the image can no longer be found by its original reference,
+			// it makes sense to show the ID instead of a stale reference.
+			newC.Image = container.ImageID
+		} else if container.ImageID == img.ID {
+			newC.Image = container.Config.Image
+		} else {
+			newC.Image = container.ImageID
+		}
```

Comment:
Need to consider if newC.Image has no tag, then should we add the default tag as it used to do?